src/state/slices/marketDataSlice/marketDataSlice.tsx

```diff
@@ -133,11 +138,12 @@ export const marketApi = createApi({
           return { error }
         }
       },
+      keepUnusedDataFor: 5, // Invalidate cached asset market data after 5 seconds.
```

Comment:
i feel like this is masking a different problem - why was this necessary?

Reply:
The default cache validity period for data fetched through RTK Query endpoints [is 60 seconds](https://redux-toolkit.js.org/rtk-query/usage/cache-behavior). Looking at the [Osmosis frontend](https://app.osmosis.zone/pool/1), it looks like the pool composition is updated more frequently than that, so the price data should change on timescales < 60s as well. 

With the default cache behavior in place, In the event that the user clicked an Osmosis opportunity, closed the modal before confirming a transaction, then opened it again within 60 seconds, the cached price data would be returned, which may very well be stale by that time. I think this is a reasonable case to account for because I ran into it myself while building out the feature.